Cake.Recipe/Content/teamcity.cake

```diff
@@ -41,7 +41,13 @@ public class TeamCityRepositoryInfo : IRepositoryInfo
 {
     public TeamCityRepositoryInfo(ITeamCityProvider teamCity, ICakeContext context)
     {
-        Branch = context.Environment.GetEnvironmentVariable("vcsroot.branch").Replace("refs/heads/", string.Empty);
+        var branchName = context.Environment.GetEnvironmentVariable("vcsroot.branch");
+        if(string.IsNullOrEmpty(branchName)) 
+        {
+            throw new Exception(@"Environment variable ""vcsroot.branch"" could not be found. Cake.Recipe needs ""vcsroot.branch"" exposed as a build parameter.");
```

Comment:
I am not certain if throwing an exception would be the best option.
Maybe falling back to trying to resolve the branch using `git` would be better (with a warning reported during the build of course).

Thoughts?

Reply:
A second thing, we may perhaps want to try to use what other tools that Cake.Recipe make use of.
In this context, I know that the `codecov-exe` expects the environment variable `TEAMCITY_BUILD_BRANCH` so we probably would want to check that as well.
https://github.com/codecov/codecov-exe#teamcity

Reply:
I added the `Exception` to get a meaningful text to the code that currently breaks with some `ArgumentNullException`,
but I agree. Finding a "better" solution is preferred.
However, I would currently not dig too deep in alternative ways to find `vcsroot.branch`, since Cake 1.1.0 will (thanks to https://github.com/cake-build/cake/pull/3113) introduce `TeamCity.Environment.Build.VcsBranchName` and `TeamCity.Environment.Build.BranchName`